Add custom clang-format, re-format all code

civerachb-cpr opened the pull request:
Nobody liked the default code formatting, so use PickNik's formatting instead.

civerachb-cpr committed:
Add custom clang-format, re-format all code

civerachb-cpr committed:
Update alignment.

mhosmar-cpr commented on puma_motor_driver/src/multi_puma_node.cpp:
```diff
@@ -48,104 +45,88 @@ MultiPumaNode::MultiPumaNode(const std::string node_name)
+  joint_names_      = this->get_parameter("joint_names").as_string_array();
+
+  RCLCPP_INFO(this->get_logger(), "Gear Ratio %f\nEncoder CPR %d\nFrequency %d\nGain PID %f,%f,%f\nCANBus Device %s",
+              gear_ratio_, encoder_cpr_, freq_, gain_p_, gain_i_, gain_d_, canbus_dev_.c_str());
```
If we like how the arguments were on different lines before we could add: `BinPackParameters: OnePerLine`

civerachb-cpr replied:
I just tried this and it didn't seem to make any difference.

Setting it to `false` seems to do what we want though.

civerachb-cpr committed:
Don't align declarations, don't pack parameters

ghost commented:
I still prefer Aligning assignment and declarations.  We could set them to Consecutive so it isn't aligned across empty lines. Ideally we would have a comment for each distinct block of assignments/declarations which would also start a new alignment block.

We could see what the rest of the team thinks

ghost commented:
Also rather than setting BinPackParameters to false setting BinPackParameters to OnePerLine will allow multiple params to be on one line if they fit but do one per line if they don't. disabling it all together will have them always on different lines.

Correction: I thought you were setting **BinPackArguments** to false.  It seems setting BinPackParameters to false is having a similar effect.  But I still think its better to set it to OnePerLine

civerachb-cpr commented:
I tried setting these, but they didn't actually make any difference:
```
AlignConsecutiveAssignments: Consecutive
AlignConsecutiveDeclarations: Consecutive
...
BinPackParameters: OnePerLine
```

I'm ambivalent about aligning declarations & assignments. Frequently they're different types anyway, so it doesn't bother me too much if they're not aligned.

civerachb-cpr committed:
Re-enable aligning consecutive declarations & assignments

tonybaltovski commented on puma_motor_driver/CMakeLists.txt:
```diff
@@ -42,13 +42,12 @@ install(TARGETS multi_puma_node
-  # a copyright and license is added to all source files
-  set(ament_cmake_cpplint_FOUND TRUE)
+  list(APPEND AMENT_LINT_AUTO_EXCLUDE
+    ament_cmake_copyright
```
Does this pass if enabled?

civerachb-cpr replied:
No:
```
test 1
    Start 1: copyright

1: Test command: /usr/bin/python3 "-u" "/opt/ros/jazzy/share/ament_cmake_test/cmake/run_test.py" "/home/civerachb/Code/clearpath_common_ws/build/puma_motor_driver/test_results/puma_motor_driver/copyright.xunit.xml" "--package-name" "puma_motor_driver" "--output-file" "/home/civerachb/Code/clearpath_common_ws/build/puma_motor_driver/ament_copyright/copyright.txt" "--command" "/opt/ros/jazzy/bin/ament_copyright" "--xunit-file" "/home/civerachb/Code/clearpath_common_ws/build/puma_motor_driver/test_results/puma_motor_driver/copyright.xunit.xml"
1: Working Directory: /home/civerachb/Code/clearpath_common_ws/src/clearpath_motor_drivers/puma_motor_driver
1: Test timeout computed to be: 200
1: -- run_test.py: invoking following command in '/home/civerachb/Code/clearpath_common_ws/src/clearpath_motor_drivers/puma_motor_driver':
1:  - /opt/ros/jazzy/bin/ament_copyright --xunit-file /home/civerachb/Code/clearpath_common_ws/build/puma_motor_driver/test_results/puma_motor_driver/copyright.xunit.xml
1: include/puma_motor_driver/can_proto.hpp: copyright=Texas Instruments Incorporated.  All rights reserved. (2008-2013), license=<unknown>
1: include/puma_motor_driver/driver.hpp: copyright=Clearpath Robotics, Inc., All rights reserved. (2015), license=<unknown>
1: include/puma_motor_driver/multi_puma_node.hpp: copyright=Clearpath Robotics, Inc., All rights reserved. (2024), license=<unknown>
1: src/driver.cpp: copyright=Clearpath Robotics, Inc., All rights reserved. (2015), license=<unknown>
1: src/multi_puma_node.cpp: copyright=Clearpath Robotics, Inc., All rights reserved. (2024), license=<unknown>
1: 5 errors, checked 5 files
1: -- run_test.py: return code 1
1: -- run_test.py: verify result file '/home/civerachb/Code/clearpath_common_ws/build/puma_motor_driver/test_results/puma_motor_driver/copyright.xunit.xml'
1/4 Test #1: copyright ........................***Failed    0.59 sec
```

mhosmar-cpr replied:
Even if we added a "clearpath" copyright type the can_proto file would fail. There might be a way to set the test to pass if any copyright is found.

mhosmar-cpr replied:
It doesn't look like you can... So the best option would be to copy what otto did and also exclude the TI file.

mikehosmar committed:
clang-format 18.1.3 compatible

mikehosmar committed:
Updates to match with clang 19

Turns out the clang-format version bundled with vscode is 19.  I was using an extension which needs it to be installed separately.

mikehosmar committed:
Allow a few cases to be on one line.
